### PR TITLE
Allow two finger touchpad scrolling on iPadOS

### PR DESCRIPTION
**Pull request by robertherber**
## Description

Adding support for touchpad in PanGestureHandler by setting `allowedScrollTypesMask` to `UIScrollTypeMaskAll` for  all devices with iOS 13.4+. Important for iPads with Touchpads and should work with Mac Catalyst through the same API as well. As described in #1100 - this covers the ipadOS part of that issue.

## Test plan

Run on an iPad with touchpad. It's now possible to two-finger swipe to interact with the PanGestureHandler.

<details>
<summary>Show code</summary>

```jsx
import React from 'react';
import { AppRegistry, SafeAreaView, Animated } from 'react-native';
import { PanGestureHandler, State } from 'react-native-gesture-handler';

const Example = () => {
  const translateX = new Animated.Value(0);
  const translateY = new Animated.Value(0);
  const lastOffset = { x: 0, y: 0 };

  const onPanGestureEvent = Animated.event([
    {
      nativeEvent: {
        translationX: translateX,
        translationY: translateY,
      },
    },
  ]);

  const onHandlerStateChange = ({
    nativeEvent: { translationX, translationY, oldState },
  }) => {
    if (oldState === State.ACTIVE) {
      // compensate a start of pan
      lastOffset.x += translationX;
      lastOffset.y += translationY;
      translateX.setOffset(lastOffset.x);
      translateY.setOffset(lastOffset.y);

      translateX.setValue(0);
      translateY.setValue(0);
    }
  };

  return (
    <SafeAreaView style={styles.container}>
      <PanGestureHandler
        onGestureEvent={onPanGestureEvent}
        onHandlerStateChange={onHandlerStateChange}>
        <Animated.View
          style={[
            styles.rectangle,
            {
              transform: [{ translateX }, { translateY }],
            },
          ]}
        />
      </PanGestureHandler>
    </SafeAreaView>
  );
};

const styles = {
  container: { flex: 1, justifyContent: 'center', alignItems: 'center' },
  rectangle: {
    backgroundColor: 'pink',
    width: 100,
    height: 100,
  },
};

AppRegistry.registerComponent('Example', () => Example);
```

</details>

### CODE REVIEW

**Comment by jakub-gonet** [comment]
This breaks CI.

**Comment by robertherber** [comment]
> This breaks CI.

I'm guessing it's because the XCode used in the CI is 10.2 (current is 11.7) and this flag requires iOS 13 (which should required XCode 11).

**Comment by jakub-gonet** [comment]
I finally bumped CI and if checks will pass I'll merge this one.

**Comment by jakub-gonet** [comment]
One question though - should we do that by default?

**Comment by robertherber** [comment]
> One question though - should we do that by default?

I think in most cases this would be the expected default behaviour but there might be some edge case I'm not aware of.

File: DrawerLayout.d.ts [hunk 1]
```diff
@@ -36,6 +36,7 @@ declare module 'react-native-gesture-handler/DrawerLayout' {
     statusBarAnimation?: StatusBarAnimation;
     overlayColor?: string;
     contentContainerStyle?: StyleProp<ViewStyle>;
+    iosEnableTouchpadTwoFingerGesture?: boolean;
```
Comment (jakub-gonet) [inline DrawerLayout.d.ts:39]: I'm not sure if we should prefix this property with `ios`. For example in Buttons we have [`activeOpacity`](https://docs.swmansion.com/react-native-gesture-handler/docs/component-buttons#activeopacity-ios-only) which is ios only or `borderless` which is android specific.

File: DrawerLayout.js [hunk 2]
```diff
@@ -50,6 +50,7 @@ export type PropType = {
   drawerContainerStyle?: any,
   contentContainerStyle?: any,
   onGestureRef?: Function,
+  iosEnableTouchpadTwoFingerGesture?: Boolean,
```
Comment (jakub-gonet) [inline DrawerLayout.js:53]: I think this should be `boolean`

File: ios/Handlers/RNPanHandler.m [hunk 3]
```diff
@@ -206,6 +206,16 @@ - (void)configure:(NSDictionary *)config
   APPLY_FLOAT_PROP(failOffsetYStart);
   APPLY_FLOAT_PROP(failOffsetYEnd);
   
+  if (@available(iOS 13.4, *)) {
+    id prop = config[@"iosEnableTouchpadTwoFingerGesture"];
+    if(prop != nil){
+      bool iosEnableTouchpadTwoFingerGesture = [RCTConvert BOOL:prop];
```
Comment (jakub-gonet) [inline ios/Handlers/RNPanHandler.m:212]: I think we can convert in the same line with `config[(...)]` if `[RCTConvert BOOL:nil]` returns `nil`.

**Review by jakub-gonet — COMMENTED** [review]
Left some inline comments.

File: DrawerLayout.d.ts (hunk 1 above)
Comment (robertherber) [reply DrawerLayout.d.ts]: Changed it to `enableTrackpadTwoFingerGesture`

File: ios/Handlers/RNPanHandler.m (hunk 3 above)
Comment (robertherber) [reply ios/Handlers/RNPanHandler.m]: Updated and works

**Comment by jakub-gonet** [comment]
I pasted some test code into a description. I don't have access to the touchpad connected to the iPad, you please verify that it works with this PR?

**Comment by robertherber** [comment]
@jakub-gonet I've verified that the test code works - and of course the examples. Ready to merge?

**Review by jakub-gonet — APPROVED** [review]
LGMT, thanks for the patch.

**Pull request merged**